src/from_geo.rs

```diff
@@ -65,61 +62,96 @@ impl<'a> TryInto<GGeom<'a>> for &'a LineString<f64> {
 struct LineRing<'a>(&'a LineString<f64>);
 
 /// Convert a geo_types::LineString to a geos LinearRing
-/// a LinearRing should be closed so cloase the geometry if needed
-impl<'a, 'b> TryInto<GGeom<'b>> for &'a LineRing<'b> {
-    type Err = Error;
+/// a LinearRing should be closed so close the geometry if needed
+/// What happens if the linestring is empty?
+impl<'a, 'b> TryFrom<&'a LineRing<'b>> for GGeom<'b> {
+    type Error = Error;
 
-    fn try_into(self) -> Result<GGeom<'b>, Self::Err> {
-        let points = &(self.0).0;
-        let nb_points = points.len();
+    fn try_from(linering: &'a LineRing<'b>) -> Result<GGeom<'b>, Self::Error> {
+        let nb_points = linering.0.num_coords();
         if nb_points > 0 && nb_points < 3 {
             return Err(Error::InvalidGeometry(
                 "impossible to create a LinearRing, A LinearRing must have at least 3 coordinates"
                     .into(),
             ));
         }
 
+        if nb_points == 0 {
+            let coords = CoordSeq::new(0, CoordDimensions::TwoD)?;
+            return GGeom::create_linear_ring(coords);
+        }
+
+        let mut points = linering.0.points_iter();
+
+        // The following expect is OK because we took care of the case where there is no points
+        let first = points.next().expect("At least one point");
+
+        // This expect is OK too, because if there is at least one point, there is at least 3 points
+        // because of the constraint above.
+        let last = points.last().expect("No last point");
+
         // if the geom is not closed we close it
-        let is_closed = nb_points > 0 && points.first() == points.last();
+        let is_closed = nb_points > 0 && first == last;
         // Note: we also need to close a 2 points closed linearring, cf test closed_2_points_linear_ring
         let need_closing = nb_points > 0 && (!is_closed || nb_points == 3);
```

Comment:
oh 😅 

> There is one test that's not working... closed_2_points_linear_ring. I don't understand what needs to be tested, and, from the little I read, it seems the diagram did not match what the code was doing.... So this point (pun intended) needs a bit more work.

I guess this is what you were talking about. =)

Reply:
I guess that, to be mathematically consistent, the application of the function that transforms a linestring into a linearing, must be applicable multiple time without an error. Let me explain....
- if i have an input linestring (shape1) that is valid according to the rules of the transformation, then it produces a linearing (shape 2). And if i apply that transformation again on shape2, then it should be valid (because it's a valid linearring) and not change it (it should produce something that is identical to shape2).

Reply:
To make sure I'm understanding, I think you're saying:

```
// provided shape1 is reasonable, and the unwrap doesn't panic...
let lr1 = LineRing::try_from(&shape1).unwrap();
let lr2 = LineRing::try_from(&lr1).unwrap();

// ...then lr1 and lr2 should always be eq.
assert_eq!(lr1, lr2);
```

Am I understanding correctly - or are you saying something else?

That would makes sense to me, but I'm not sure why you mention it. Is the example I gave susceptible to breaking that contract?